lib/Sema/CSApply.cpp

```diff
@@ -3885,8 +3885,10 @@ namespace {
     }
 
     Expr *visitLazyInitializerExpr(LazyInitializerExpr *expr) {
-      simplifyExprType(expr);
-      assert(expr->getType()->isEqual(expr->getSubExpr()->getType()));
+      auto type = simplifyType(cs.getType(expr));
+      cs.setType(expr, type);
```

Comment:
I don't think you want to do cs.setType() here.

Reply:
I'm not sure I follow - original cs.getType returns a type variable,  then it gets "simplified" to a concrete type and gets reset in constraint system, this is a common pattern across all CSApply e.g. `simplifyExprType` does that too, all of the types get set to expressions at the end.

Reply:
Then why not call simplifyExprType() here?

Reply:
Yes, I'm to quite sure anymore why I changed it...

Reply:
Ah yes, because if I were to use `simplifyExprType` I'd have to query cache twice for the type, but i guess it might be okay to do